week-14/homework/index.js

```diff
@@ -0,0 +1,157 @@
+    prevBtn.disabled = false;
+    filter.disabled = false;
+}
+
+let renderFilter = function (genresData) {
+    let allGenres = genresData.data;
+
+    //render filter dropdown
+    filter.innerHTML = '';
+    if (currentFilter == 'top') {
+        filter.innerHTML += `<option selected value="top">Top anime</option>`;
+    } else {
+        filter.innerHTML += `<option value="top">Top anime</option>`;
+    }
+    for (let i = 0; i < allGenres.length; i++) {
+        if (currentFilter == allGenres[i].mal_id) {
+            filter.innerHTML += `<option selected value="${allGenres[i].mal_id}">${allGenres[i].name} (${allGenres[i].count})</option>`;
+        } else {
+            filter.innerHTML += `<option value="${allGenres[i].mal_id}">${allGenres[i].name} (${allGenres[i].count})</option>`;
+        }
+    }
+}
+
+//fetch from api
+let fetchData = function () {
+    //remove d-none to show loader and disable buttons while loading
+    loader.classList.remove('d-none');
+    selectPagination.disabled = true;
+    nextBtn.disabled = true;
+    prevBtn.disabled = true;
+    filter.disabled = true;
+
+    //add loading animation
+    let allListItems = list.querySelectorAll('div');
+    for (let i = 0; i < allListItems.length; i++) {
+        allListItems[i].classList.add('animate-pulse');
+    }
+
+    //Checks if current filter is 'top' else renders genres
+    if (currentFilter == 'top') {
+        fetch(`https://api.jikan.moe/v4/top/anime?page=${currentPage}`).
+            then(function (response) {
+                return response.json();
+            }).then(renderData);
+    } else {
+        fetch(`https://api.jikan.moe/v4/anime?genres=${currentFilter}&page=${currentPage}`).
+            then(function (response) {
+                return response.json();
+            }).then(renderData);
+    }
+
+    //Renders the filter
+    fetch('https://api.jikan.moe/v4/genres/anime').
+        then(function (response) {
+            return response.json();
+        }).then(renderFilter);
+}
+
+let loadPreviousPage = function () {
+    if (currentPage == 1) {
+    } else {
+        currentPage--;
+        fetchData();
+    }
+}
+
+let loadNextPage = function () {
+    if (currentPage >= totalPages) {
+    } else {
+        currentPage++;
+        fetchData();
+    }
+}
+
+let selectPage = function () {
+    let selectedPage = selectPagination.value;
+    if (selectedPage == currentPage) {
+    } else {
+        currentPage = selectedPage;
+        fetchData();
+    }
+}
+
+let selectFilter = function () {
+    let selectedFilter = filter.value;
+    if (selectedFilter == currentFilter) {
+    } else {
+        currentPage = 1;
+        currentFilter = selectedFilter;
+        fetchData();
+    }
+}
+//add event listeners
+prevBtn.addEventListener('click', loadPreviousPage);
+nextBtn.addEventListener('click', loadNextPage);
+selectPagination.addEventListener('click', selectPage);
+filter.addEventListener('click', selectFilter);
```

Comment:
Same here, this is `select` so we should use `addEventListener('change',...)` instead